src/solver/optimisation/opt_gestion_des_couts_cas_lineaire.cpp

```diff
@@ -41,32 +42,50 @@
 #include "spx_constantes_externes.h"
 
 static void shortTermStorageCost(
-  const ::ShortTermStorage::AREA_INPUT& shortTermStorageInput,
-  const CORRESPONDANCES_DES_VARIABLES* CorrespondanceVarNativesVarOptim,
+  int PremierPdtDeLIntervalle,
+  int DernierPdtDeLIntervalle,
+  int NombreDePays,
+  const std::vector<::ShortTermStorage::AREA_INPUT>& shortTermStorageInput,
+  CORRESPONDANCES_DES_VARIABLES** CorrespondanceVarNativesVarOptim,
   double* linearCost)
 {
-    for (auto& storage : shortTermStorageInput)
+    SIM::SpreadGenerator spreadGenerator;
+    int pdtJour = 0;
+    for (int pays = 0; pays < NombreDePays; ++pays)
     {
-        const int clusterGlobalIndex = storage.clusterGlobalIndex;
-        if (const int varLevel
-            = CorrespondanceVarNativesVarOptim->SIM_ShortTermStorage.LevelVariable[clusterGlobalIndex];
-            varLevel >= 0)
+        for (const auto& storage : shortTermStorageInput[pays])
         {
-            linearCost[varLevel] = 0;
-        }
+            // We use the global index as a seed to avoid identical draws for multiple storages
+            // which may lead to equivalent solutions if 2 storages are identical
+            spreadGenerator.reset(storage.clusterGlobalIndex);
```

Comment:
About : `spreadGenerator.reset(storage.clusterGlobalIndex);`
Really sorry but I still do not understand why we need to reset the seed for each ST storage.
Maybe I miss something but, if we use the same spreadGenerator for each ST storage and we just increment the spread without resetting the seed, we'll have different hourly spreads for each ST storage, won't we ?

Reply:
Yes, I agree. Let's drop this !